src/app.ts

```diff
@@ -26,45 +26,62 @@ program
 program
   .command('build')
   .description('Builds the report into a folder named "dist"')
-  .action(() => {
+  .action(async () => {
     const cliConfig = loadCliConfig();
-    new Builder(console).build(cliConfig.distPath, cliConfig.buildCommand).catch((err) => {
-      console.error(chalk.red(err));
-    });
+    const builder = new Builder(console);
+
+    try {
+      await builder.build(cliConfig.distPath, cliConfig.buildCommand);
+    } catch (error) {
+      console.error(chalk.red(error));
+    }
   });
 
 program
   .command('upload')
   .description('Bundles and uploads the report to the configured workspace')
-  .action(() => {
-    console.log(chalk.yellow(chalk.italic('Bundling and uploading your project...')));
+  .action(async () => {
     const cliConfig = loadCliConfig();
     const lxrConfig = loadLxrConfig();
     const url = `https://${lxrConfig.host}/services/pathfinder/v1/reports/upload`;
-    new Builder(console)
-      .build(cliConfig.distPath, cliConfig.buildCommand)
-      .then(() => new Uploader().upload(url, lxrConfig.apitoken, lxrConfig.host, lxrConfig.proxyURL))
-      .catch(handleError);
+
+    const builder = new Builder(console);
+    const uploader = new Uploader();
+
+    console.log(chalk.yellow(chalk.italic('Bundling and uploading your project...')));
+
+    try {
+      await builder.build(cliConfig.distPath, cliConfig.buildCommand);
+      await uploader.upload(url, lxrConfig.apitoken, lxrConfig.host, lxrConfig.proxyURL);
+    } catch (error) {
+      handleError(error);
+    }
   });
 
 program
   .command('store-upload <id> <apitoken>')
   .description('Bundles and uploads the report to the LeanIX Store')
   .option('--host <host>', 'Which store to use (default: store.leanix.net)')
   .option('--tokenhost <tokenhost>', 'Where to resolve the apitoken (default: app.leanix.net)')
-  .action((id: string, apitoken: string, options: { host: string; tokenhost: string }) => {
+  .action(async (id: string, apitoken: string, options: { host: string; tokenhost: string }) => {
     const cliConfig = loadCliConfig();
 
     const host = options.host || 'store.leanix.net';
     const tokenhost = options.tokenhost || 'app.leanix.net';
     const msg = `Bundling and uploading your project to the LeanIX Store (${host})...`;
+    const url = `https://${host}/services/torg/v1/assetversions/${id}/payload`;
+
+    const builder = new Builder(console);
+    const uploader = new Uploader();
+
     console.log(chalk.yellow(chalk.italic(msg)));
```

Comment:
I guess we could use the `msg` inline as well here, to make it consistent.

Reply:
Good catch, looks cleaner now 😃